### DISCUSSION TITLE
TuYa TS0001_switch_module as momentary switch

### DISCUSSION THREAD

**Discussion by bubune** [Koenkk/zigbee2mqtt#19767] [Devices]
Hi, I've juste bought TuYa TS0001_switch_module to command portail. To do it, I've juste need an impulsion of 1 sec about once is pressed or command via Home Assistant. So I want activate the relay and it will close automatically after a second later. I do not want do this stuff in Home Assistant as it's not the right place to do it. By buying this switch, I've belived that setup can be done by setting the switch as 'momentary' in zigbee2Mqtt, but it is not working. So I am wondering if it is possible or not? If possible, what should I do to make it possible? Thanks

**Comment by sauveron** [comment]
I have the same issue. Have you solved it?

**Comment by bubune** [comment]
Hi,

this switch doesn't work like you expect it.
I bought this one, 1ch
https://fr.aliexpress.com/item/1005005195039361.html?spm=a2g0o.order_list.order_list_main.5.2e785e5bPpa0RL&gatewayAdapt=glo2fra

Regards,
Bruno

**Comment by gfrantsau** [comment]
It's very disappointing that momentary type can actually be selected in the settings but doesn't work. Hopefully @Koenkk will add support for this function.

**Comment by Matssa56** [comment]
Damn I wish I saw this post before, momentum doesn't work on my side either and it's the only small dry relay on ZigBee that I found (other than the dual channel Aqara). Untill it's fixed, I'm gonna do it directly in home assistant via mqtt but if the option worked in Z2M it would be awesome.

**Comment by KelArth** [comment]
Hi. Any news about the fix ? got OXT one.

**Reply by gogolores** [reply]
Nope, no fix yet. Abandon all hope ...
